File: Java/ImageEditor/src/Driver.java

/**
 * Created by andrei on 6/13/15.
 */
public class Driver {

    private final static String INVERT = "invert";
    private final static String GRAYSCALE = "grayscale";
    private final static String EMBOSS = "emboss";
    private final static String MOTION_BLUR = "motion_blur";

    public static void main(String[] args) {
        String fileName = "TestFiles/audio.ppm";
        String action = "motion_blur".toLowerCase();
        int motionValue = 20;
        ImageProcessor imageProcessor = new ImageProcessor(fileName);
        imageProcessor.setMotionValue(motionValue);
        imageProcessor.processContent();
        switch (action) {
            case INVERT:
                imageProcessor.invert();
                break;
            case GRAYSCALE:
                imageProcessor.grayScale();
                break;
            case EMBOSS:
                imageProcessor.emboss();
                break;
            case MOTION_BLUR:
                imageProcessor.motionBlur();
                break;
            default:
                System.out.println("case doesn't exist");
        }
    }
}

File: CPP/ImageEditor/ImageEditor/driver.cpp
//
//  main.cpp
//  ImageEditor
//
//  Created by Andrei Rybin on 6/20/15.
//  Copyright (c) 2015 Andrei Rybin. All rights reserved.
//

#include <iostream>
#include "ImageProcessor.h"


int main(int argc, const char * argv[]) {
    
    std::string INVERT = "invert";
    std::string GRAYSCALE = "grayscale";
    std::string EMBOSS = "emboss";
    std::string MOTIONBLUR = "motionblur";
    int motionValue = 20;


    std::string fileName = "TestFiles/audio.ppm";
    std::string action = "invert";
    ImageProcessor iProcessor(fileName);
    iProcessor.processContent();
    if(action.compare(INVERT) == 0)
    {
        iProcessor.invert();
    }
    else if(action.compare(GRAYSCALE) == 0)
    {
        iProcessor.grayscale();
    }
    else if(action.compare(EMBOSS) == 0)
    {
        iProcessor.emboss();
    }
    else if(action.compare(MOTIONBLUR) == 0)
    {
        iProcessor.setMotionValue(motionValue);
        iProcessor.motionblur();
    }
    
    
    return 0;
}
